dev/PowerNotifications/ClosedSourceMock.cpp

```diff
@@ -0,0 +1,271 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include <pch.h>
+#include <unordered_map>
+#include <PowerManager.g.h>
+#include <PowerNotificationsPal.h>
+
+using namespace std;
+using index = ULONGLONG;
+index g_idx;
+
+unordered_map<index, OnCompositeBatteryStatusChanged> onCompositeBatteryStatusChanged_callbacks;
+unordered_map<index, OnDischargeTimeChanged> onDischargeTimeChanged_callbacks;
+unordered_map<index, OnEnergySaverStatusChanged> onEnergySaverStatusChanged_callbacks;
+unordered_map<index, OnPowerConditionChanged> onPowerConditionChanged_callbacks;
+unordered_map<index, OnDisplayStatusChanged> onDisplayStatusChanged_callbacks;
+unordered_map<index, OnSystemIdleStatusChanged> onSystemIdleStatusChanged_callbacks;
+unordered_map<index, OnPowerSchemePersonalityChanged> onPowerSchemePersonalityChanged_callbacks;
+unordered_map<index, OnUserPresenceStatusChanged> onUserPresenceStatusChanged_callbacks;
+unordered_map<index, OnSystemAwayModeStatusChanged> onSystemAwayModeStatusChanged_callbacks;
+
+HRESULT GetCompositeBatteryStatus(CompositeBatteryStatus** compositeBatteryStatusOut)
```

Comment:
The argument type here should probably have been CompositeBatteryStatus* instead. For a fixed-size struct outparam, we shouldn't need to have the implementation heap-allocate it; we should be able to have the caller allocate it and the implementation fill it. This pattern with CoTaskMemAlloc is more useful for dynamically-sized or opaque outparams, where the caller can't appropriately initialize an empty outparam for the implementation to fill.